index.bs

```diff
@@ -1610,7 +1610,7 @@ Create a named {{MLOperand}} based on a descriptor, that can be used as an input
     </div>
     1. If |name| is empty, then [=exception/throw=] a {{TypeError}}.
     1. [=Assert=]: the type of |descriptor| is {{MLOperandDescriptor}}.
-    1. [=Assert=]: If |descriptor|.{{MLOperandDescriptor/dimensions}} does not [=map/exist=], then |descriptor| defines a scalar input, equivalent to zero dimensions.
+    1. [=Assert=]: If |descriptor|.{{MLOperandDescriptor/dimensions}} does not [=map/exist=], then |descriptor| defines a scalar input, and |descriptor|.{{MLOperandDescriptor/dimensions}} is set to an empty array of {{unsigned long long}}.
```

Comment:
The dimensions setting operation looks like an algorithm step. I am not sure whether we should put it into an assertion. @inexorabletash, any insights?

Reply:
Agreed, this doesn't make sense as an assertion. But it also doesn't make sense as a step, since the following step is "if descriptor.dimensions exists:" which would never be false if this step set it to something. If this is intending to describe the behavior non-normatively, then a NOTE might be appropriate to explain subtle behavior. I'm not sure what's intended, so I'd revisit the whole algorithm to make sure it's coherent.

Also, at this stage in the processing, _dimensions_ would not be an array (a JS concept) or a sequence (an IDL concept) but a [list](https://infra.spec.whatwg.org/#lists) (from Infra).

Reply:
Thanks for clarifying! I think this is what you want as text:

> 1. If name is empty, then throw a TypeError.
> 2. If descriptor.dimensions does not exist, then set descriptor.dimensions to an empty list.
> 3. If the check dimensions steps given descriptor.dataType and descriptor.dimensions return false, then throw a "DataError" DOMException.
> 4. If the byte length of descriptor is not supported by the underlying platform, then throw a "DataError" DOMException.
> 5. If any of the following sub-steps fail....

(With all the markdown added back in!)

In case it isn't clear, the _"make an effective copy of the tensor description"_ is implicit with the spec processing model - the JS object passed as the MLOperandDescriptor dictionary is analyzed and all the fields are extracted into some implementation-specific representation (e.g. a C++ object), so the descriptor object is already a copy with modifications unobservable by script.

(Re: sequence/array/list - agreed, separate PR makes sense - cleanup-only PRs are easier to review and understand in the repo history!)

Reply:
(Note that in my example text I dropped the assertion about the input type that's currently step 2, tracked in #455 but it's harmless to leave it in for now.)

Reply:
Thank you for the explicit snippets and links.

Reply:
@huningxin Ningxin, if you're out for Lunar New Year (🏮🎇) until the 24th, I'm considering completing this since I believe all feedback is addressed now, and I can create another CR for any smaller new feedback.